src/fmu/pem/pem_functions/fluid_properties.py

```diff
@@ -70,11 +74,28 @@ def _adjust_bubble_point(
             gas_gravity=oil_gas_gravity,
             temperature=temp,
         )
-        idx_below = pres <= bp
+        # Only consider cells where the oil saturation is positive
+        idx_below = np.logical_and(pres <= bp, so > 0.0)
         if np.any(idx_below):
+            # More than 1% of cells below bubble point will be regarded as an error
+            # situation if the gas Z-factor is set to default value of 1.0. If the
+            # user modifies the Z-factor, we regard cells below bubble point to be
+            # an expected situation and not an error.
+            frac_below = np.sum(idx_below) / pres.size
+            if frac_below > BUBBLE_POINT_FRACTION_TOLERANCE and isclose(
+                zone.gas_z_factor, 1.0
+            ):
+                raise ValueError(
+                    f"Fraction of cells with pressure below oil bubble point is "
+                    f"{frac_below:.3f}. "
+                    "If this is an expected situation, add a "
+                    "gas Z-factor (deviation from an ideal gas) to the YAML parameter "
+                    "file for each PVTNUM zone, e.g.: 'gas_z_factor: 0.97' "
+                    "The gas Z-factor must deviate from 1.0."
```

Comment:
Do we want to use this error message to also ask them to notify the maintainers of `fmu-pem`, in order for them to learn if there are actual "valid" use cases for many cells below bubble point, or if the cases where this occur is fixable in practice before `fmu-pem` enters the scene? Could motivate them actually notifying maintainers by saying this property/workaround might be removed in future if we see that this escape hatch / "post-fix" of the model should not be needed?